### PR TITLE
pantheon.elementary-mail: init at unstable-2021-06-21

### LINKED ISSUE #83252: Package pantheon's mail client
**Project description**
The [elementary mail client](https://github.com/elementary/mail) shares a history with [geary](https://wiki.gnome.org/Apps/Geary), but is a different app, which fits the pantheon desktop.

**Metadata**

* homepage URL: https://elementary.io/
* source URL: https://github.com/elementary/mail
* license: GPLv3
* platforms: Linux

@worldofpeace I think you're coordinating pantheon efforts? I'd be happy to take a look at this.

### PR DESCRIPTION
**Pull request by ethancedwards8**
###### Motivation for this change

Closes https://github.com/NixOS/nixpkgs/issues/83252
Aids in https://github.com/NixOS/nixpkgs/issues/58161

The reasoning for choosing an unstable version is that there hasn't been a release in ***3 years***. My hope is they'll "release" soon (the project is still seeing very active development) and we can move to it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/pantheon/apps/elementary-mail/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "elementary-mail";
+  version = "unstable-2021-06-21";
+
+  repoName = "mail";
+
+  src = fetchFromGitHub {
+    owner = "elementary";
+    repo = repoName;
```
Comment (SuperSandro2000) [inline pkgs/desktops/pantheon/apps/elementary-mail/default.nix:34]:
```suggestion
  src = fetchFromGitHub {
    owner = "elementary";
    repo = "mail";
```

Comment (ethancedwards8) [reply pkgs/desktops/pantheon/apps/elementary-mail/default.nix:34]: All the other files do this, I did it this way to keep it uniform.

Comment (ethancedwards8) [reply pkgs/desktops/pantheon/apps/elementary-mail/default.nix:34]: See:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/pantheon/apps/elementary-photos/default.nix#L40-L46
https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/pantheon/apps/elementary-photos/default.nix#L40-L47
Sure its a bit weird, and perhaps unnecessary, but I think all the applications should stay uniform.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**